src/net.h

```diff
@@ -308,19 +308,25 @@ class Transport {
         const std::string& /*m_type*/
     >;
 
-    /** Get bytes to send on the wire.
+    /** Get bytes to send on the wire, if any, along with other information about it.
      *
      * As a const function, it does not modify the transport's observable state, and is thus safe
      * to be called multiple times.
      *
-     * The bytes returned by this function act as a stream which can only be appended to. This
-     * means that with the exception of MarkBytesSent, operations on the transport can only append
-     * to what is being returned.
+     * @param[in] have_next_message controls whether the "more" return value indicates more bytes
+     *            to be sent before (have_next_message=false) or after (have_next_message=true) a
+     *            potential SetMessageToSend immediately afterwards. It is set by the caller when
+     *            they know they have another message ready to send. The have_next_message
+     *            argument only affects this "more" return value and nothing else.
```

Comment:
d67be9ad3836bd19cc8ea71a0b99c7d60682dd32: alternative explanation, which I think is more clear (but perhaps wrong):

> have_next_message: the caller knows it has another message ready to send. The have_next_message argument (only) affects the "more" return value, which is used to set the `MSG_MORE` TCP flag and to trigger optimistic send. With v2 transport more can be false even if vSendMsg has more messages left. This happens before the handshake is complete.

I found this part confusing: "indicates more bytes to be sent before (have_next_message=false) or after (have_next_message=true) a potential SetMessageToSend immediately afterwards"